charm_versions.md

```diff
@@ -0,0 +1,138 @@
+string.
+
+In the past, the version would be set automatically if the charm was deployed from a known version
+control system, but that is no longer the case. The intention is that charmers set this version to
+allow tracking the installed version of the charm back to the source tree that it was built from.
+
+### Charm Revision
+
+All charms have a revision, which is a simple positive integer. You can see the revisions available
+of a charm published to Charmhub with `charmcraft revisions <charm name>`, and in the drop-down near
+the top of a charm's page in Charmhub. You can see the installed revision of a charm in
+`juju status` in the application table in the "Rev" column in the tabular view, or in the
+`applications.<app name>.charm-rev` field in JSON or YAML format.
+
+The revision is automatically set when uploading the charm using `charmcraft upload`. Note that
+revisions may be unreleased, or may be released only for specific
+[channels](https://juju.is/docs/sdk/charmcraft-release).
+
+When a charm is deployed locally, Juju will use an initial revision of 1 and then increment the
+revision each time a new version of the same charm is deployed or refreshed.
+
+### Libraries
+
+Charm libs have both an "API version" and a "patch version". Any changes to the lib API that are not
+backwards compatible will be made in a new API version, while new patch versions include compatible
+improvements including new features, general improvements, and bug fixes.
+
+The available versions for a charm lib can be found on Charmhub in the "Libraries" tab. These can
+also be found using `charmcraft list-lib <charm name>`.
+
+The API and patch versions of a charm lib are contained within the lib source file itself. Search
+for `LIBAPI` (the API version) and `LIBPATCH` (the patch version) within the source.
+
+It is not possible to see which version of a lib was used in a deployed charm, other than by using
+the `charm-version` to examine the source tree from which the charm was built or retrieving the
+deployed charm source from a unit. In general, the lib version is a charm author concern, rather
+than one of an admin.
+
+Authors publish new versions of charm libraries using the
+[charmcraft publish-lib command](https://juju.is/docs/sdk/create-and-publish-a-charm-library#heading--update-a-charm-library),
+setting the [metadata](https://juju.is/docs/sdk/library#heading--structure) appropriately - in
+particular, the `LIBAPI` value for the API version (which is also in the path) and the `LIBPATCH`
+value for the patch version.
+
+To upgrade the revision of a lib used by a charm, the same `charmcraft fetch-lib` command is used
+as when originally installing the lib. To upgrade the version of the lib, use `fetch-lib` but
+change the version number in the path, for example from `charms.demo.v0.demo` to
+`charms.demo.v1.demo`. See the
+[documentation](https://juju.is/docs/sdk/find-and-use-a-charm-library#heading--update-a-charm-library)
+for more details.
+
```

Comment:
We already have https://juju.is/docs/sdk/library and https://juju.is/docs/sdk/manage-libraries . This should be incorporated there (if the details aren't there already).

PS Note that starting with Charmcraft 2.5 charmers are only supposed to edit a single yaml file directly -- the charmcraft.yal file. Every other necessary yaml file in a charm project is generated automatically during `charmcraft pack`.